### PR TITLE
ECS cluster for proxy

### PR DESCRIPTION
**Pull request by aboulay**
- Rework network module to add proxy subnets (public and private).
- Split IAM JSON generator in three folder.
- Add module for ECS deployment.
- Add module for RDS deployment.
- Add first version of secret management.
- Add module for proxy deployment.
- Add dns record for mainnet.tezoslink.io.

### CODE REVIEW

File: infra/terraform/modules/network/proxy_private.tf [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+  vpc_id = aws_vpc.tzlink.id
+
+  tags = {
+    Name        = format("tzlink-%s-private-proxy-a", var.ENV)
+    Project     = var.PROJECT_NAME
+    Environment = var.ENV
+    BuildWith   = var.BUILD_WITH
+  }
+
+  # route {
```
Comment (louptheron) [inline infra/terraform/modules/network/proxy_private.tf]: Why commented ?

File: infra/terraform/modules/network/proxy_private.tf [hunk 2]
```diff
@@ -0,0 +1,85 @@
+################### proxy A ######################
```
Comment (louptheron) [inline infra/terraform/modules/network/proxy_private.tf:1]: Add that this is about subnets ? i.e `private_subnets` or something like that

File: infra/terraform/modules/network/proxy_private.tf [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+    Name        = format("tzlink-%s-private-proxy-b", var.ENV)
+    Project     = var.PROJECT_NAME
+    Environment = var.ENV
+    BuildWith   = var.BUILD_WITH
+  }
+
+  # route {
+  #   cidr_block = "0.0.0.0/0"
+  #   gateway_id = aws_nat_gateway.public_proxy_b.id
+  # }
```
Comment (louptheron) [inline infra/terraform/modules/network/proxy_private.tf]: Same

File: infra/terraform/modules/network/proxy_public.tf [hunk 4]
```diff
@@ -0,0 +1,61 @@
+################### Public A ######################
```
Comment (louptheron) [inline infra/terraform/modules/network/proxy_public.tf:1]: This is also about subnet, and the backend API may be located here non ?

File: infra/vaulted.tfvars [hunk 5]
```diff
@@ -0,0 +1,6 @@
+$ANSIBLE_VAULT;1.1;AES256
```
Comment (louptheron) [inline infra/vaulted.tfvars:1]: Could you add to the README how the passwords are encrypted  and how to use that ?

File: infra/terraform/modules/network/proxy_private.tf (hunk 2 above)
Comment (aboulay) [reply infra/terraform/modules/network/proxy_private.tf:1]: private_subnets is in the title of the file. This is just to identify where start the private_subnet_a (for proxy) and the private_subnet_b (for proxy).

File: infra/terraform/modules/network/proxy_public.tf (hunk 4 above)
Comment (aboulay) [reply infra/terraform/modules/network/proxy_public.tf:1]: Currently, the public_subnet exists to keep the internet facing loadbalancers. I think that I'll keep that name for the moment and I'll made the refacto when Ill work on api.

File: infra/vaulted.tfvars (hunk 5 above)
Comment (aboulay) [reply infra/vaulted.tfvars:1]: I can but Ill not let the password in the README. I'm on it.

File: infra/terraform/modules/network/proxy_private.tf (hunk 1 above)
Comment (aboulay) [reply infra/terraform/modules/network/proxy_private.tf]: Oups, Ill remove this !

File: infra/terraform/modules/network/proxy_private.tf (hunk 3 above)
Comment (aboulay) [reply infra/terraform/modules/network/proxy_private.tf]: Same than before, Ill remove it soon.

**Review by louptheron — APPROVED** [review]

**Pull request merged**